src/mongo_odm/test/query_builder.cpp

```diff
@@ -961,6 +1021,19 @@ TEST_CASE("Update Builder", "mongo_odm::update_expr") {
 
     /* Array update operators */
 
+    SECTION("Test $ array update operator.", "[mongo_odm::dollar_operator_nvp]") {
+        auto res =
+            coll.update_one((MONGO_ODM_KEY(Bar::w) == 555,
+                             MONGO_ODM_KEY(Bar::arr).elem_match(MONGO_ODM_ELEM(Bar::arr) == 5)),
+                            MONGO_ODM_KEY(Bar::arr).update_first() = 500);
```

Comment:
Thoughts on renaming this to `first_match()`? `update_first` is a little confusing.